src/conda_devenv/devenv.py

```diff
@@ -375,28 +386,20 @@ def merge_dependencies_version_specifications(
                 package_name = dep
                 package_version = ""
             else:
-                m = re.match(package_pattern, dep, flags=re.IGNORECASE)
-                if m is None:
-                    raise RuntimeError(
-                        'The package version specification "{}" do not follow the'
-                        " expected format.".format(dep)
-                    )
+                package = PackageSpecifier.parse(dep)
                 # Consider the channel name as part of the package name.
                 # If multiple channels are specified, the package will be repeated.
-                package_name = m.group("package")
-                if m.group("channel"):
-                    package_name = m.group("channel") + package_name
-
-                package_version = m.group("version")
+                package_name = package.channel + package.name
+                package_version = package.version
 
             # OrderedDict is used as an ordered set, the value is ignored.
             version_matchers = new_dependencies.setdefault(
                 package_name, collections.OrderedDict()
             )
             if len(package_version) > 0:
-                version_matchers[package_version] = True
+                version_matchers[package_version] = None
```

Comment:
Just to understand better, why did this change?

Reply:
Oh this is minor, just that to me, `None` conveys better the notion of "the value is ignored" than `True`. 😝